graylog2-server/src/main/java/org/graylog2/security/MongoDbSessionDAO.java

```diff
@@ -107,8 +118,23 @@ protected void doUpdate(Session session) {
         } else {
             throw new RuntimeException("Unsupported session type: " + session.getClass().getCanonicalName());
         }
-
-        mongoDBSessionService.saveWithoutValidation(dbSession);
+        // Due to https://jira.mongodb.org/browse/SERVER-14322 upserts can fail under concurrency.
+        // We need to retry the update, and stagger them a bit, so no all of the retries attempt it at the same time again.
+        // Usually this should succeed the first time, though
+        final Retryer<Object> retryer = RetryerBuilder.newBuilder()
```

Comment:
Actually, `<Void>newBuilder()` and then also return a `Void` in the callable. I think it makes it less clear what's happening, so I'd vote for leaving it like it is.